generator/mavgen_c.py

```diff
@@ -384,7 +384,9 @@ def generate_message_h(directory, m):
  */
 static inline ${return_type} mavlink_msg_${name_lower}_get_${name}(const mavlink_message_t* msg${get_arg})
 {
-    return _MAV_RETURN_${type}${array_tag}(msg, ${array_return_arg} ${wire_offset});
+    if (msg->len > ${wire_offset})
+        return _MAV_RETURN_${type}${array_tag}(msg, ${array_return_arg} ${wire_offset});
+    return (${return_type}) 0;
```

Comment:
```suggestion
    if (msg->len <= ${wire_offset}) {
        // zero-bytes have been truncated from this message
        return (${return_type}) 0;
    }
    return _MAV_RETURN_${type}${array_tag}(msg, ${array_return_arg} ${wire_offset});
```

The braces because that's the style used throughout this file, even on single-line bodies - and that many bugs have been introduced by not having them :-)

I reordered to do early return and to make a nice place to put an extra comment about what's going on.

Reply:
I agree, that is better solution. Patch updated.